variants/particle_xenon/variant.cpp

```diff
@@ -27,33 +27,43 @@
 const uint32_t g_ADigitalPinMap[] =
 {
   // D0 .. D8
-  26,   // P0.26 (SDA)
-  27,   // P0.27 (SCL)
-  33,   // P1.01 (SDA1) (UART1_RTS) (SPI1_SCK) (PWM3)
-  34,   // P1.02 (SCL1) (UART1_CTS) (SPI1_MOSI) (PWM3)
-  40,   // P1.08 (UART2_TX) (SPI1_MISO) (PWM1)
-  42,   // P1.10 (UART2_RX) (PWM1)
-  43,   // P1.11 (UART2_CTS) (PWM1)
-  44,   // P1.12 (PWM0)
-  35,   // P1.03 (UART2_RTS)
+  _PINNUM(0, 26),   // P0.26 (SDA)
+  _PINNUM(0, 27),   // P0.27 (SCL)
+  _PINNUM(1, 1),    // P1.01 (SDA1) (UART1_RTS) (SPI1_SCK) (PWM3)
+  _PINNUM(1, 2),    // P1.02 (SCL1) (UART1_CTS) (SPI1_MOSI) (PWM3)
+  _PINNUM(1, 8),    // P1.08 (UART2_TX) (SPI1_MISO) (PWM1)
+  _PINNUM(1, 10),   // P1.10 (UART2_RX) (PWM1)
+  _PINNUM(1, 26),   // P1.11 (UART2_CTS) (PWM1)
+  _PINNUM(1, 26),   // P1.12 (PWM0)
+  _PINNUM(1, 26),   // P1.03 (UART2_RTS)
 
   // D9 .. D20
-  6,    // P0.06 (UART1_TX)
-  8,    // P0.27 (UART1_RX)
-  46,   // P1.14 (SPI_MISO)
-  45,   // P1.13 (SPI_MOSI)
-  47,   // P1.15 (SPI_SCK)
-  31,   // P0.31 (PWM3) (SPI_SS) (ADC5)
-  30,   // P0.30 (PWM3) (ADC4)
-  29,   // P0.29 (PWM2) (ADC3)
-  28,   // P0.28 (PWM2) (ADC2)
-  4,    // P0.04 (PWM2) (ADC1)
-  3,    // P0.03 (PWM2) (ADC0)
-  11,   // P0.11 (MODE)
-  18,   // P0.18 (RESET)
+  _PINNUM(0, 6),    // P0.06 (UART1_TX)
+  _PINNUM(0, 8),    // P0.08 (UART1_RX)
+  _PINNUM(1, 14),   // P1.14 (SPI_MISO)
+  _PINNUM(1, 13),   // P1.13 (SPI_MOSI)
+  _PINNUM(1, 15),   // P1.15 (SPI_SCK)
+  _PINNUM(0, 31),   // P0.31 (PWM3) (SPI_SS) (ADC5)
+  _PINNUM(0, 30),   // P0.30 (PWM3) (ADC4)
+  _PINNUM(0, 29),   // P0.29 (PWM2) (ADC3)
+  _PINNUM(0, 28),   // P0.28 (PWM2) (ADC2)
+  _PINNUM(0, 4),    // P0.04 (PWM2) (ADC1)
+  _PINNUM(0, 3),    // P0.03 (PWM2) (ADC0)
+  _PINNUM(0, 11),   // P0.11 (MODE)
+  _PINNUM(0, 18),   // P0.18 (RESET)
 };
 
-static void switch_antenna(bool useExternal) {
+void initVariant()
+{
+  switch_antenna(false);
+
+  led_pwm_init(LED_PRIMARY_IDX, LED_PRIMARY_PIN);
```

Comment:
Error during compilation for missing function "led_pwm_init"

Reply:
I'm going to see if there's a way to incorporate this with analogWrite somehow